File: 71.simplify-path.py
#
# @lc app=leetcode id=71 lang=python3
#
# [71] Simplify Path
#

# @lc code=start
class Solution:
    def simplifyPath(self, path: str) -> str:
        result=[]
        direntries =  path.split("/")
        while "" in direntries:
            direntries.remove("")
        while "." in direntries:
            direntries.remove(".")
        for direntry in direntries:
            if direntry == "..": 
                if len(result):
                    result.pop()
            else:
                result.append(direntry)
        return "/"+"/".join(result)
        
# @lc code=end

if __name__ =="__main__":
        
    print(Solution().simplifyPath("/a/./b/../../c/"))

File: golang/71.simplify-path.go
/*
 * @lc app=leetcode.cn id=71 lang=golang
 *
 * [71] Simplify Path
 */
package main

import "strings"

// @lc code=start
type stack struct {
	data []string
}

func newStack() *stack {
	return &stack{}
}
func (s *stack) Push(item string) {
	s.data = append(s.data, item)
}
func (s *stack) Pop() {
	if len(s.data) == 0 {
		return
	}
	s.data = s.data[:len(s.data)-1]
}

func simplifyPath(path string) string {
	s := newStack()
	pathArray := strings.Split(path, "/")
	for _, item := range pathArray {
		// for //
		if item == "" || item == "." {
			continue
		}
		if item == ".." {
			s.Pop()
		} else {
			s.Push(item)
		}

	}
	return "/" + strings.Join(s.data, "/")
}

// @lc code=end
